extensions/wsc/src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java

```diff
@@ -0,0 +1,165 @@
+import org.mule.runtime.extension.api.annotation.metadata.Content;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataKeyId;
+import org.mule.runtime.extension.api.annotation.metadata.OutputResolver;
+import org.mule.runtime.extension.api.annotation.metadata.TypeResolver;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.runtime.operation.OperationResult;
+import org.mule.runtime.module.xml.stax.StaxSource;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+
+import net.sf.saxon.jaxp.SaxonTransformerFactory;
+import org.apache.cxf.attachment.AttachmentImpl;
+import org.apache.cxf.binding.soap.SoapHeader;
+import org.apache.cxf.endpoint.Client;
+import org.apache.cxf.message.Attachment;
+import org.apache.cxf.message.Exchange;
+import org.apache.cxf.message.ExchangeImpl;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+
+/**
+ * The only {@link WebServiceConsumer} operation which consumes an operation of the connected web service and returns it's
+ * response.
+ * <p>
+ * The consume operation expects an XML body and a set of headers and attachments if required.
+ * <p>
+ * For the cases where no input parameters are required the {@link RequestBodyGenerator} will generate a body to perform the
+ * operation if a {@code null} value is passed.
+ *
+ * @since 4.0
+ */
+public class ConsumeOperation {
+
+  public static final String MULE_ATTACHMENTS_KEY = "mule.wsc.attachments";
+  public static final String MULE_HEADERS_KEY = "mule.wsc.headers";
+
+  @OutputResolver(output = OutputBodyResolver.class, attributes = WscAttributesResolver.class)
+  public OperationResult<String, WscAttributes> consume(@Connection WscConnection connection,
+                                                        @MetadataKeyId(OperationKeysResolver.class) String operation,
+                                                        @Optional @Content @TypeResolver(InputBodyResolver.class) String body,
+                                                        @Optional @TypeResolver(InputHeadersResolver.class) Map<String, String> headers,
+                                                        @Optional List<WscAttachment> attachments)
+      throws Exception {
+    connection.addOutInterceptor(new SoapActionInterceptor(operation));
```

Comment:
this is wrong. The connections are pooled but you add this interceptor everytime. You either have to not pool them or make sure to remove this interceptor.